infra/utils.py

```diff
@@ -89,14 +89,11 @@ def get_fuzz_targets(path):
 
 def get_container_name():
   """Gets the name of the current docker container you are in.
-  /proc/self/cgroup can be used to check control groups e.g. Docker.
-  See: https://docs.docker.com/config/containers/runmetrics/ for more info.
 
   Returns:
     Container name or None if not in a container.
   """
-  with open('/proc/self/cgroup') as file_handle:
-    if 'docker' not in file_handle.read():
+  if b'docker' not in subprocess.run('systemd-detect-virt -c', shell=True, stdout=subprocess.PIPE).stdout:
```

Comment:
Thank you so much! you rock!

can you please do
result = subprocess.run('systemd-detect-virt -c', shell=True, stdout=subprocess.PIPE)
if b'docker' in result.stdout:

Reply:
stdout=subprocess.PIPE can be skipped as well i think

Reply:
`subprocess.run(['systemd-detect-virt', '-c']).stdout` should work.

Reply:
According to https://docs.python.org/3/library/subprocess.html#subprocess.run, `run` doesn't capture stdout by default. Another option would be to pass `capture_output=True` to it but it was added relatively recently so I don't expect it to be available on Ubuntu (where python-3.5 is still used).

Reply:
Why not just subprocess.check_output
This is equivalent to:
run(..., check=True, stdout=PIPE).stdout

Reply:
`systemd-detect-virt -c` returns 1 when it isn't run in containers, which makes `check_output` throws exceptions left and right. I don't think it's desirable given that that code can be run elsewhere.

Reply:
ok then, please move this to a local var, that line is too long. rest lgtm.

Reply:
I replaced `shell=True` with `['systemd-detect-virt', '-c']` as @jonathanmetzman suggested.

I'll add `result=...` shortly